git_pile/git_pile.py

```diff
@@ -1372,13 +1372,13 @@ def init(self):
     def run(self):
         args = self.args
         config = self.config
-        assert_required_tools()
-        assert_format_patch_compatible_args(args)
-
+        root = config.normalize(git_root_or_die())
         if not config.check_is_valid():
             return 1
 
-        root = git_root_or_die()
+        assert_required_tools()
```

Comment:
Unrelated comment: I wonder if `assert_required_tools() ` shouldn't be called at the very beginning of the program (except for the handling of --version and --help)...

Reply:
I will leave this for another time